CustomCollections/CustomCollections/CustomVector.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CustomCollections
+{
+    class CustomVector<T> : IList<T>
+    {
+        private T[] _vector;
+
+        public int Count { get; private set; } = 0;
+        public int Length { get; private set; } = 1;
+        public bool IsReadOnly { get; } = false;
```

Comment:
This one is part of ICollection, but noone will ever use it when working with Vector, because Vector is always mutable. This is good candidate for explicit implementation.